dotnetv3/S3/SSECLowLevelMPUcopyObjectExample/SSECLowLevelMPUcopyObjectExample/SSECLowLevelMPUcopyObject.cs

```diff
@@ -0,0 +1,142 @@
+
+namespace SSECLowLevelMPUcopyObjectExample
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Security.Cryptography;
+    using System.Threading.Tasks;
+    using Amazon.S3;
+    using Amazon.S3.Model;
+
+    /// <summary>
+    /// Uses the Amazon Simple Storage Service (Amazon S3) low level API to
+    /// perform a multi-part upload to an Amazon S3 bucket. The example was
+    /// created using the AWS SDK for .NET version 3.7 and .NET Core 5.0.
+    /// </summary>
+    public class SSECLowLevelMPUcopyObject
+    {
+        public static async Task Main()
+        {
+            string existingBucketName = "doc-example-bucket";
+            string sourceKeyName = "sample_file.txt";
+            string targetKeyName = "sample_file_copy.txt";
+            string filePath = $"sample\\{targetKeyName}";
+
+            // If the AWS region defined for your default user is different
+            // from the AWS regsion where your Amazon S3 bucket is located,
+            // pass the AWS Region name to the S3 client object's constructor.
+            // For example, RegionEndpoint.USWest2.
+            IAmazonS3 client = new AmazonS3Client();
+
+            // Create the encryption key.
+            var base64Key = CreateEncryptionKey();
+
+            await CreateSampleObjUsingClientEncryptionKeyAsync(
+                client, existingBucketName,
+                sourceKeyName, filePath, base64Key);
+        }
+
+        /// <summary>
+        /// Creates the encryption key to use with the multi-part upload.
+        /// </summary>
+        /// <returns>A string containing the base64 key for encrypting the
+        /// multi-part upload.</returns>
+        public static string CreateEncryptionKey()
+        {
+            Aes aesEncryption = Aes.Create();
+            aesEncryption.KeySize = 256;
+            aesEncryption.GenerateKey();
+            string base64Key = Convert.ToBase64String(aesEncryption.Key);
+            return base64Key;
+        }
+
+        /// <summary>
+        /// Creates and uploads an object using a multi-part upload.
+        /// </summary>
+        /// <param name="client">The initialized Amazon S3 object used to
+        /// initialize and perform the multi-part upload.</param>
+        /// <param name="existingBucketName">The name of the bucket to which
+        /// the object will be uploaded.</param>
+        /// <param name="sourceKeyName">The source object name.</param>
+        /// <param name="filePath">The location of the source object.</param>
+        /// <param name="base64Key">The encryption key to use with the upload.</param>
+        public static async Task CreateSampleObjUsingClientEncryptionKeyAsync(
+            IAmazonS3 client,
+            string existingBucketName,
+            string sourceKeyName,
+            string filePath,
+            string base64Key)
+        {
+            // List to store upload part responses.
```

Comment:
The list

Reply:
When I looked at this one, it seems that anyone who knows C# would know his is a generic List of Responses. I removed the comment altogether.